base_layer/core/src/base_node/states/forward_block_sync.rs

```diff
@@ -237,13 +237,12 @@ async fn download_blocks<B: BlockchainBackend + 'static>(
     match shared.comms.fetch_blocks_with_hashes(curr_headers.clone()).await {
         Ok(blocks) => {
             info!(target: LOG_TARGET, "Received {} blocks from peer", blocks.len());
-
-            if let StatusInfo::BlockSync(ref mut info) = shared.info {
-                info.tip_height = Some(blocks[blocks.len() - 1].block().header.height);
-                info.local_height = Some(blocks[0].block().header.height);
-            }
-            shared.publish_event_info().await;
             for i in 0..blocks.len() {
+                if let StatusInfo::BlockSync(ref mut info) = shared.info {
+                    info.tip_height = Some(blocks[blocks.len() - 1].block().header.height);
+                    info.local_height = Some(blocks[0].block().header.height + (i as u64));
+                }
+                shared.publish_event_info().await;
```

Comment:
Did you mean to publish this `blocks.len()` times? - I think it's better to just add the `!blocks.is_empty()` check above the loop

Reply:
Wel it depends on the accuracy you want the status to be shown. If you want it to only update once per batch of blocks updated, then it should be above the for loop. 
If not then you need to publish this channel every block. 

Its easy to update it both ways, I am not particular to one side or the other

Reply:
Agree. Rather just publish the status update once.

Reply:
cwl, updated